neo4j/internal/bolt/bolt6_test.go

```diff
@@ -0,0 +1,1975 @@
+					Method:      s,
+				},
+				testCase{
+					description:    "warning minimum severity",
+					MinSev:         notifications.WarningLevel,
+					ExpectedMinSev: &warningSev,
+					Method:         s,
+				},
+				testCase{
+					description:     "disabled categories",
+					DisCats:         notifications.DisableCategories(notifications.Unsupported, notifications.Generic),
+					ExpectDisCats:   true,
+					ExpectedDisCats: []any{"UNSUPPORTED", "GENERIC"},
+					Method:          s,
+				},
+				testCase{
+					description:     "warning minimum severity and disabled categories",
+					MinSev:          notifications.WarningLevel,
+					DisCats:         notifications.DisableCategories(notifications.Unsupported, notifications.Generic),
+					ExpectDisCats:   true,
+					ExpectedDisCats: []any{"UNSUPPORTED", "GENERIC"},
+					ExpectedMinSev:  &warningSev,
+					Method:          s,
+				},
+				testCase{
+					description:     "disable no categories",
+					DisCats:         notifications.DisableNoCategories(),
+					ExpectDisCats:   true,
+					ExpectedDisCats: []any{},
+					Method:          s,
+				})
+		}
+		inner.Parallel()
+		for _, test := range testCases {
+			inner.Run(fmt.Sprintf("%s for %s", test.description, test.Method), func(t *testing.T) {
+				bolt, cleanup := connectToServer(t, func(srv *bolt6server) {
+					go func() {
+						srv.waitForHandshake()
+						srv.acceptManifestVersion()
+						offerings := []protocolVersion{
+							{major: 6, minor: 0, back: 0},
+							{major: 5, minor: 8, back: 8},
+							{major: 4, minor: 4, back: 2},
+						}
+						srv.sendManifestOfferings(offerings)
+						major, minor := srv.waitForManifestConfirmation()
+						if major != 6 || minor != 0 {
+							panic(fmt.Sprintf("Expected client to choose Bolt 6.0, but got %d.%d", major, minor))
+						}
+						srv.waitForHelloWithoutAuthToken()
+						srv.acceptHello()
+						srv.waitForLogon()
+						srv.acceptLogon()
+						fieldAssertion := func(fieldNum int) func(fields []any) {
+							return func(fields []any) {
+								if test.ExpectedMinSev != nil {
+									AssertStringEqual(t, fields[fieldNum].(map[string]any)["notifications_minimum_severity"].(string), *test.ExpectedMinSev)
+								} else {
+									AssertMapDoesNotHaveKey(t, fields[fieldNum].(map[string]any), "notifications_minimum_severity")
+								}
+								// For Bolt 6, the key is 'notifications_disabled_classifications'
+								if test.ExpectDisCats {
+									AssertDeepEquals(t, fields[fieldNum].(map[string]any)["notifications_disabled_classifications"], test.ExpectedDisCats)
+								} else {
+									AssertMapDoesNotHaveKey(t, fields[fieldNum].(map[string]any), "notifications_disabled_classifications")
+								}
+							}
+						}
+						if test.Method == "run" {
+							srv.waitForRun(fieldAssertion(2))
+						} else {
+							srv.waitForTxBegin(fieldAssertion(0))
+						}
+						srv.sendFailureMsg("Neo.ClientError.Statement.SyntaxError", "Syntax error")
+					}()
```

Comment:
```suggestion
					srv.acceptBolt6WithManifest()
					fieldAssertion := func(fieldNum int) func(fields []any) {
						return func(fields []any) {
							if test.ExpectedMinSev != nil {
								AssertStringEqual(t, fields[fieldNum].(map[string]any)["notifications_minimum_severity"].(string), *test.ExpectedMinSev)
							} else {
								AssertMapDoesNotHaveKey(t, fields[fieldNum].(map[string]any), "notifications_minimum_severity")
							}
							// For Bolt 6, the key is 'notifications_disabled_classifications'
							if test.ExpectDisCats {
								AssertDeepEquals(t, fields[fieldNum].(map[string]any)["notifications_disabled_classifications"], test.ExpectedDisCats)
							} else {
								AssertMapDoesNotHaveKey(t, fields[fieldNum].(map[string]any), "notifications_disabled_classifications")
							}
						}
					}
					if test.Method == "run" {
						srv.waitForRun(fieldAssertion(2))
					} else {
						srv.waitForTxBegin(fieldAssertion(0))
					}
					srv.sendFailureMsg("Neo.ClientError.Statement.SyntaxError", "Syntax error")
```
Suggested changes:
 * Call `acceptBolt6WithManifest` instead of inlining it.
 * No need to wrap the goroutine in a goroutine

Reply:
Fixed 👍